### PR TITLE
Feature/placement dates

### PR DESCRIPTION
**Pull request by peteryates**
### Context

Schools need to be able to manage their fixed placement dates

### Changes proposed in this pull request

This set of changes provides a _very basic_ admin interface for adding, deleting and editing placement dates.

### Guidance to review

Ensure changes look suitable and sensible

### CODE REVIEW

File: app/models/bookings/school.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@ class Bookings::School < ApplicationRecord
     class_name: "Bookings::SchoolType",
     foreign_key: :bookings_school_type_id
 
+  has_one :school_profile,
+    class_name: "Schools::SchoolProfile",
+    foreign_key: :bookings_school_id
```
Comment (rjlynch) [inline app/models/bookings/school.rb]: Might be worth adding a dependent destroy here

File: app/models/schools/school_profile.rb [hunk 2]
```diff
@@ -130,6 +132,14 @@ class SchoolProfile < ApplicationRecord
       through: :college_phase_subjects,
       dependent: :destroy
 
+    has_many :placement_dates,
+      class_name: 'Bookings::PlacementDate',
+      foreign_key: :schools_school_profile_id
```
Comment (rjlynch) [inline app/models/schools/school_profile.rb]: Also dependent destroy is probably good here too

File: db/migrate/20190417081149_add_bookings_school_id_to_schools_school_profile.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddBookingsSchoolIdToSchoolsSchoolProfile < ActiveRecord::Migration[5.2]
+  def change
+    add_column :schools_school_profiles, :bookings_school_id, :integer, null: false
+    add_foreign_key :schools_school_profiles, :bookings_schools
```
Comment (rjlynch) [inline db/migrate/20190417081149_add_bookings_school_id_to_schools_school_profile.rb]: is it worth adding an index on here for the bookings_school_id?

Comment (peteryates) [reply db/migrate/20190417081149_add_bookings_school_id_to_schools_school_profile.rb]: Yep, done

**Pull request merged**